tests/ipex/test_modeling_causal_lm.py

```diff
@@ -62,43 +62,61 @@ class IPEXModelForCausalLMTest(unittest.TestCase):
     GENERATION_LENGTH = 100
     SPEEDUP_CACHE = 1.0
 
-    @parameterized.expand(SUPPORTED_ARCHITECTURES)
-    def test_compare_to_transformers(self, model_arch):
+    @parameterized.expand(
+        grid_parameters(
+            {
+                "model_arch": SUPPORTED_ARCHITECTURES,
+                "prepare_inputs": [True, False],
+            }
+        )
+    )
+    def test_compare_to_transformers(self, test_name, model_arch, prepare_inputs):
         model_id = MODEL_NAMES[model_arch]
         set_seed(SEED)
         dtype = torch.float16 if IS_XPU_AVAILABLE else torch.float32
         ipex_model = IPEXModelForCausalLM.from_pretrained(model_id, torch_dtype=dtype, device_map=DEVICE)
         self.assertIsInstance(ipex_model.config, PretrainedConfig)
         tokenizer = AutoTokenizer.from_pretrained(model_id)
-        tokens = tokenizer(
-            "This is a sample",
-            return_tensors="pt",
-            return_token_type_ids=False if model_arch in ("llama2",) else None,
-        ).to(DEVICE)
-        inputs = ipex_model.prepare_inputs_for_generation(**tokens)
-        outputs = ipex_model(**inputs)
+        texts = ["This is a sample", ["This is the first input", "This is the second input"]]
+        for text in texts:
+            tokens = tokenizer(
+                text,
+                return_tensors="pt",
+                return_token_type_ids=False if model_arch in ("llama2",) else None,
+            ).to(DEVICE)
+            if prepare_inputs:
+                inputs = ipex_model.prepare_inputs_for_generation(**tokens)
+                outputs = ipex_model(**inputs)
+            else:
+                outputs = ipex_model(**tokens)
```

Comment:
both case can be included at the same time so that we don't have to re-export the model in both case with something like 
```suggestion
            outputs = ipex_model(**tokens)
            inputs = ipex_model.prepare_inputs_for_generation(**tokens)
            outputs_2 = ipex_model(**inputs)
            self.assertTrue(torch.allclose(outputs.logits, outputs_2.logits, atol=1e-3))
```